### ISSUE TITLE
Adapt font size to screen dpi or size

### ISSUE THREAD

**Issue by orontee** [orontee/taranis#89]
# Context

Cf. #86 

<img src="https://github.com/orontee/taranis/assets/2065954/6b7b3cca-a95a-4a2e-9119-04ff31cf02a3" width=300/>

# Analysis

Font size is currently hard-coded, see https://github.com/orontee/taranis/blob/main/src/fonts.h#L7.

It could be automatically selected, may be depending on display dpi? Or, since there is a limited number of possible screen sizes, there could be fixed font size for each screen size:
```
#define HWC_DISPLAY_6INCH (HWC_DISPLAY == 0)
#define HWC_DISPLAY_5INCH (HWC_DISPLAY == 1)
#define HWC_DISPLAY_8INCH (HWC_DISPLAY == 2)
#define HWC_DISPLAY_9INCH (HWC_DISPLAY == 3)
#define HWC_DISPLAY_6INCH_758 (HWC_DISPLAY == 4) // display 1024x758 6" HD
#define HWC_DISPLAY_8TRITON (HWC_DISPLAY == 5) // display 1800x800 8"
#define HWC_DISPLAY_8INCH_1200 (HWC_DISPLAY == 6) // display 1600x1200 8"
#define HWC_DISPLAY_4INCH_480_800 (HWC_DISPLAY == 8) // display 480x800 4.3"
#define HWC_DISPLAY_6INCH_1072_1448 (HWC_DISPLAY == 9) // display 1448x1072 6"
```
Even more simpler, there are "themes". Maybe using current theme fonts would solve our problem?

**Comment by orontee** [comment]
On my e-reader the default font from the theme is really tiny, thus I applied a scale factor arbitrarily computed from the screen DPI. On my device, this computation leads to slightly smaller font than we used to have; It improves readability. I like it.

But it must be tested on other devices! (I'll build a pre-release to help people test this feature).

Two other changes related to readability:
- Font anti-aliasing is now activated
- Dates in daily forecast are displayed with the "small" font
